server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexService.java

```diff
@@ -394,6 +395,15 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                     indexSettingsBuilder.put(IndexMetadata.SETTING_INDEX_VERSION_CREATED.getKey(), createdVersion);
                 }
 
+                var settingsToValidate = indexSettingsBuilder.build();
+                if (IndexSettings.INDEX_SOFT_DELETES_SETTING.get(settingsToValidate) == false
+                    && IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(settingsToValidate).onOrAfter(Version.V_5_0_0)) {
+                    throw new IllegalArgumentException(
+                        "Creating tables with soft-deletes disabled is no longer supported. "
+                        + "Please do not specify a value for setting [soft_deletes.enabled]."
+                    );
+                }
```

Comment:
Similar to what @mfussenegger commented on an older PR state, can we move this into a setting validator instead?

Reply:
The main reason i put this on the master node is that the index version is required to be set before, which happens right before this code block as part of the index creation. However, i will have a second look.